pkg/resources/object_renaming_acceptance_test.go

```diff
@@ -0,0 +1,1366 @@
+			databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+			newDatabaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+			schemaName := acc.TestClient().Ids.Alpha()
+			tableName := acc.TestClient().Ids.Alpha()
+
+			databaseConfigModel := model.Database("test", databaseId.Name())
+			databaseConfigModelWithNewId := model.Database("test", newDatabaseId.Name())
+
+			testSteps := []resource.TestStep{
+				{
+					Config: config.FromModel(t, databaseConfigModel) +
+						configSchemaWithReferences(t, databaseConfigModel.ResourceReference(), testCase.DatabaseInSchemaDependency, databaseId.Name(), schemaName) +
+						configTableWithReferences(t, databaseConfigModel.ResourceReference(), testCase.DatabaseDependency, "snowflake_schema.test", testCase.SchemaDependency, databaseId.Name(), schemaName, tableName),
+					ExpectError: testCase.ExpectedFirstStepError,
+				},
+			}
+
+			if testCase.ExpectedFirstStepError == nil {
+				testSteps = append(testSteps,
+					resource.TestStep{
+						ConfigPlanChecks: resource.ConfigPlanChecks{
+							PreApply: []plancheck.PlanCheck{
+								plancheck.ExpectResourceAction("snowflake_database.test", plancheck.ResourceActionUpdate),
+								plancheck.ExpectResourceAction("snowflake_schema.test", plancheck.ResourceActionDestroyBeforeCreate),
+								plancheck.ExpectResourceAction("snowflake_table.test", plancheck.ResourceActionDestroyBeforeCreate),
+							},
+						},
+						Config: config.FromModel(t, databaseConfigModelWithNewId) +
+							configSchemaWithReferences(t, databaseConfigModelWithNewId.ResourceReference(), testCase.DatabaseInSchemaDependency, newDatabaseId.Name(), schemaName) +
+							configTableWithReferences(t, databaseConfigModelWithNewId.ResourceReference(), testCase.DatabaseDependency, "snowflake_schema.test", testCase.SchemaDependency, newDatabaseId.Name(), schemaName, tableName),
+					},
+				)
+			}
+
+			resource.Test(t, resource.TestCase{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+					tfversion.RequireAbove(tfversion.Version1_5_0),
+				},
+				CheckDestroy: acc.CheckDestroy(t, resources.Table),
+				Steps:        testSteps,
+			})
+		})
+	}
+}
+
+func TestAcc_DeepHierarchy_AreInConfig_SchemaRenamedInternally(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableObjectRenamingTest)
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	testCases := []struct {
+		DatabaseDependency     DependencyType
+		SchemaDependency       DependencyType
+		ExpectedFirstStepError *regexp.Regexp
+	}{
+		{DatabaseDependency: ImplicitDependency, SchemaDependency: NoDependency, ExpectedFirstStepError: regexp.MustCompile("error creating table")}, // tries to create table before schema
```

Comment:
Aren't these kind of cases also non-deterministic? A schema could be created before a table.

Reply:
I also thought about it, but in this configuration, the order seems to be consistent xd. Only this one test is non-deterministic. What's even crazier is that the same test, but with a different order in the config (`_ConfigOrderSwap` test), had some impact on the execution order because it also seems to be more consistent. Don't ask me, but non-deterministic results only seem to happen in `TestAcc_ShallowHierarchy_IsInConfig_RenamedExternally_WithoutDependency_AfterRenameDatabaseReferencingOldNameAndSchemaReferencingOldDatabaseName` (I think it was this one).

Reply:
Yeah, that's really weird, but we don't need to spend much time here, because almost always we'll have dependencies in such configurations.